zapisy/apps/theses/models.py

```diff
@@ -125,6 +126,12 @@ def is_supporting_advisor_assigned(self, user):
     def is_among_advisors(self, user):
         return self.is_mine(user) or self.is_supporting_advisor_assigned(user)
 
+    def is_user_privileged_for_thesis(self, user):
+        return self.is_among_advisors(user) or user.is_staff or is_theses_board_member(user)
+
+    def is_user_allowed_to_generate_declarations(self, user):
+        return self.is_student_assigned(user) or self.user_privileged_for_thesis(user) or self.has_been_accepted
```

Comment:
W tej chwili ten warunek jest łatwiej spełnić, niż którykolwiek z dwóch obecnie istniejących, które nim zastępujemy – np. w sytuacji, kiedy temat jest już zaakceptowany, to status/tożsamość użytkownika jest bez znaczenia.

Gdyby zrobić
```suggestion
        return self.user_privileged_for_thesis(user) or (self.has_been_accepted and self.is_student_assigned(user))
```
to byłoby chyba dokładnie tak samo, jak teraz w `gen_form`. Zastępowane wyrażenie z `view_thesis` jest pozornie mniej restrykcyjne, bo nie ma tam koniunkcji z `self.has_been_accepted`, ale do tej pory niezaakceptowany temat nie był wyświetlany jego studentowi (więc cała podstrona nie była widoczna), a to właśnie zmieniamy i również tam chyba taki kształt tego warunku będzie dobry.